ipatests/create_external_ca.py

```diff
@@ -24,12 +24,18 @@
 from cryptography.hazmat.primitives.asymmetric import rsa
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
+import ipalib.constants
 
 import datetime
 import six
 
 ISSUER_CN = 'example.test'
 
+MAP_SIGNING_ALGORITHM = {
+    'SHA256withRSA': hashes.SHA256,
+    'SHA512withRSA': hashes.SHA512,
```

Comment:
``SHA512withRSA`` makes only sense for RSA keys with a very large key size. For comparison a 15360bit RSA key has about the same strength as a 256bit AES key.

``SHA348`` and RSASSA-PSS algorithms are missing.

Reply:
The idea is that we need to have a mapping place for names to hash algorithm objects. I can add more here, no problem, as long as we keep them in sync with what could be set as a constant in `ipalib.constants`.

Reply:
Specific choice for the default is irrelevant in this place.